### DISCUSSION TITLE
TSA: how are multiple observations on the same day processed?

### DISCUSSION THREAD

**Discussion by dominiqueweber** [davidfrantz/force#208] [Q&A] [answered]
Hi all

I am using Sentinel-2 and Landsat-8 data for TSA and calling a UDF there. Since there is a trade-off between spatial and temporal resolution, I wonder how multiple observations on the same day are processed? For example, if NDVI values are available for Sentinel-2 and Landsat-8 on the same day, is the average taken? Would it be possible to give preference to Sentinel-2 in such a situation to benefit from the higher spatial resolution?

Thanks for your help,
Dominique

**Comment by davidfrantz** [comment, marked as answer]
Hi @dominiqueweber,

all individual observations will be passed to your UDF (unless you used interpolation, then you will receive the interpolated values). There can be multiple observations per day (e.g. up to 3 when Landsat 7, 9 and Sentinel-2 were all active - or when working in high latitudes).

You should have the sensor information as variable in your UDF to implement an individual aggregation:
sensors:   numpy.ndarray[nDates](str)

Cheers,
David

**Reply by dominiqueweber** [reply]
Makes sense, thanks a lot David.

**Reply by davidfrantz** [reply]
You're welcome!
